File: 02-Classes/2.1.1/c/Rect.c
#include <stdio.h>

typedef struct{
    int x ,y;
    int w, h;
} Rect;

void print(Rect*r) {
    printf("Quandrado de tamanho (%d, %d) na posicao (%d, %d).\n" ,r->w, r->h, r->x, r->y);
}

void main (void) {
    Rect r1 = {1,1, 10, 10};
    print(&r1);
}

File: 02-Classes/2.3.1/Figures/Rect.java
package Figures;

import java.awt.*;

public class Rect {
    private int x, y;
    private int w, h;
    Color corDePreenchimento;
    Color corDeBorda;

    public Rect (int x, int y, int w, int h, Color corDePreenchimento) {
        this.x = x;
        this.y = y;
        this.w = w;
        this.h = h;
        this.corDePreenchimento = corDePreenchimento;
        // this.corDeBorda = corDeBorda;
    }

    public void print () {
        System.out.format("Retangulo de tamanho (%d,%d) na posicao (%d,%d).\n",
            this.w, this.h, this.x, this.y);
    }

    public void paint (Graphics g) {
        Graphics2D g2d = (Graphics2D) g;
        
        // g2d.setStroke(new BasicStroke(5));
        // g2d.setColor(corDeBorda);
        // g2d.drawRect(this.x,this.y, this.w,this.h);
        g2d.setColor(corDePreenchimento);
        g2d.fillRect(this.x, this.y, this.w , this.h);
    }
}